qiskit/qobj/pulse_qobj.py

```diff
@@ -0,0 +1,610 @@
+
+        Args:
+            data (dict): A dictionary for the experiment config
+
+        Returns:
+            PulseLibraryItem: The object from the input dictionary.
+        """
+        return cls(**data)
+
+    def __eq__(self, other):
+        if isinstance(other, PulseLibraryItem):
+            if self.to_dict() == other.to_dict():
+                return True
+        return False
+
+
+class PulseQobj:
+    """A Pulse Qobj."""
+
+    def __init__(self, qobj_id, config, experiments=None,
+                 header=None):
+        """Instatiate a new QASM Qobj Object.
+
+        Each QASM Qobj object is used to represent a single payload that will
+        be passed to a Qiskit provider. It mirrors the Qobj the published
+        `Qobj specification <https://arxiv.org/abs/1809.03452>`_ for OpenQASM
+        experiments.
+
+        Args:
+            qobj_id (str): An identifier for the qobj
+            config (PulseQobjConfig): A config for the entire run
+            header (QobjHeader): A header for the entire run
+            experiments (list): A list of lists of :class:`PulseQobjExperiment`
+                objects representing an experiment
+        """
+        self.qobj_id = qobj_id
+        self.config = config
+        self.header = header or QobjHeader()
+        self.experiments = experiments or []
+        self.type = 'PULSE'
+        self.schema_version = '1.1.0'
+
+    def to_dict(self, validate=False):
+        """Return a dictionary format representation of the Pulse Qobj.
+
+        Note this dict is not in the json wire format expected by IBMQ and qobj
+        specification because complex numbers are still of type complex. Also
+        this may contain native numpy arrays. When serializing this output
+        for use with IBMQ you can leverage a json encoder that converts these
+        as expected. For example:
+
+        .. code-block::
+
+            import json
+            import numpy
+
+            class QobjEncoder(json.JSONEncoder):
+                def default(self, obj):
+                    if isinstance(obj, numpy.ndarray):
+                        return obj.tolist()
+                    if isinstance(obj, complex):
+                        return (obj.real, obj.imag)
+                    return json.JSONEncoder.default(self, obj)
+
+            json.dumps(qobj.to_dict(), cls=QobjEncoder)
+
+        Args:
+            validate (bool): When set to true validate the output dictionary
+                against the jsonschema for qobj spec.
+
+        Returns:
+            dict: A dictionary representation of the QasmQobj object
+        """
+        out_dict = {
+            'qobj_id': self.qobj_id,
+            'header': self.header.to_dict(),
+            'config': self.config.to_dict(),
+            'schema_version': self.schema_version,
+            'type': self.type,
+            'experiments': [x.to_dict() for x in self.experiments]
+        }
+        if validate:
```

Comment:
This should be broken out to a method call.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/3383/commits/0c0f50d5fb28cacd8f82b00aaa5fe1797260b083